lib/azureService.js

```diff
@@ -0,0 +1,77 @@
+const azure = require('azure-storage');
+
+const config = require('./config');
+const sortByFilename = require('./sortByFilenameDateDesc');
+
+const blobSvc = azure.createBlobService();
+const options = {
+  clientRequestTimeoutInMs: config.azureTimeoutMinutes * 60 * 1000,
+};
+
+function listBlobs(containerName) {
+  return new Promise((resolve, reject) => {
+    blobSvc.listBlobsSegmented(containerName, null, (error, result) => {
+      if (!error) {
+        resolve(result.entries);
+      } else {
+        reject(error);
+      }
+    });
+  });
+}
+
+function uploadToAzure(containerName, filename, blobName) {
+  return new Promise((resolve, reject) => {
+    blobSvc.createBlockBlobFromLocalFile(
```

Comment:
If the container doesn't exist when an upload occurs I assume this will fail?
There is a function to create one that might be useful http://azure.github.io/azure-storage-node/#toc5__anchor

Reply:
It will, containers are generally created during initial configuration, I can add a check/create to the function if you feel it necessary.